README.rst
Description
===========

Plango is product that replaces the default Plone Icon Set with icons from the Tango Project.

During product installation, some new layers is added to all skins selections in the portal_skins tool.

The new layers corresponds to some folders in portal_skins which contains icon replacements.

The `Tango Icon Set <http://tango.freedesktop.org/Tango_Icon_Library>`_ is part of the `Tango Desktop Project <http://tango.freedesktop.org>`_.

Installation
============

Place the product folder in the Products directory of your Zope instance and restart the server.

The classic way
---------------

Go to the 'Site Setup' page in the Plone interface and click on the 'Add/Remove Products' link.

Choose the product (check its checkbox) and click the 'Install' button.

Uninstall -- This can be done from the same management screen, but only if you installed it from the quick installer.

The Generic Setup way
---------------------

In the ZMI, go to 'portal_setup' and, (1) select the 'Properties' tab and choose 'Plango' in the popup list before clicking 'Update'.
Then (2) go to the 'Import' tab and click 'Import all steps'.

While adding a 'Plone Site' to Zope (from the ZMI), you can select 'Plango' in the proposed 'Extension Profiles' to have it installed automatically during the creation of the portal.

Uninstall -- This must be done manually from the ZMI, as GenericSetup does not have an API for removing/uninstalling stuff (yet).

Author
======

Paolo Melchiorre <paulox@paulox.net>

Contributor
===========

David Convent <davconvent@gmail.com>

Credits
=======

Tango Desktop Project for Icons and `PauLoX wiki account <http://tango.freedesktop.org/User:PauLoX>`_.

License
=======

View `LICENSE`_ and `GPL`_.

.. _LICENSE: LICENSE.txt
.. _GPL: GPL.txt


HISTORY.txt
2007-03-05 Paolo Melchiorre <paulox@paulox.net>

    * Renamed directories:
        plango_mimetypes_icons, plango_ploneboard_images, plango_plone_images
    * Added icon in plango_plone_images:
        edit_rename.gif
    * Added directories:
        plango, plango/plonekupuimages
    * Added icons in plango:
        accessibility_icon.gif, contact_icon.gif, login_icon.gif
    * Added icons in plango/plonekupuimages:
        bold.gif, center.gif, definitionlist.gif, delete_icon.gif,
        hyperlink.gif, image.gif, indent.gif, italic.gif, left_just.gif,
        list.gif, numbered_list.gif, outdent.gif, redo.gif, right_just.gif,
        strikethrough.gif, subscript.gif, superscript.gif, table.gif,
        underline.gif, undo.gif, unformat.gif, view-source.gif
    * Updated images: screenshot-en.gif
    * version.txt: Updated to 2.1.3
    * Updated and formatted: HISTORY.txt, LICENSE.txt, LICENZA.it_IT.txt,
        README.it_IT.txt, README.txt.

2007-02-27 Paolo Melchiorre <paulox@paulox.net>

    * Updated images: screenshot-en.gif, screenshot-it.gif
    * version.txt: Updated to 2.1.2

2007-02-26 Paolo Melchiorre <paulox@paulox.net>

    * Added icons: book_icon.gif, error_icon.gif, error_log_icon.gif,
        info.gif, save_icon.gif, undo_icon.gif
    * Updated icons: info_icon.gif
    * Added directory: tango_mimetypes_icons
    * Added icons in tango_mimetypes_icons:
        application.png, c.png, down.png, f.png, image.png, log.png,
        mime.png, pl.png, py.png, sh.png, sxi.png, tex.png, unknown.png,
        video.png, audio.png, cpp.png, dvi.png, gf.png, info.png, man.png,
        o.png, png.png, quicktime.png, sxc.png, sxm.png, text.png, up.png,
        wav.png, avi.png, deb.png, exe.png, h.png, iso.png, message.png,
        pdf.png, ppt.png, rar.png, sxd.png, sxw.png, tgz.png, vcal.png,
        xls.png, core.png, doc.png, font.png, html.png, java.png, midi.png,
        pk.png, ps.png, rpm.png, sxg.png, tar.png, txt.png, vcard.png

2007-02-23 Paolo Melchiorre <paulox@paulox.net>

    * Updated images: screenshot-en.gif, screenshot-it.gif
    * Updated icons: discussionitem_icon.gif, user.gif, group.gif, rss.gif
    * Added icons: phone_icon.gif, lock_icon.gif
    * Added icons in tango_ploneboard_images:
        hot_icon.gif, lock_icon.gif, new_conversation.gif, norm_icon.gif,
        ploneboard_comment_icon.gif, ploneboard_conversation_icon.gif,
        ploneboard_forum_icon.gif, ploneboard_icon.gif, smiley_cool.png,
        smiley_lol.png, smiley_sad.png, smiley_skeptic.png, smiley_smile.png,
        smiley_surprised.png, smiley_tongue.png, smiley_wink.png, sticky_icon.gi
    * version.txt: Updated to 2.1.1

2007-02-16 Paolo Melchiorre <paulox@paulox.net>

    * Extensions/Install.py: bug fixed
    * *: UTF8 issue fixed

2007-02-15 Paolo Melchiorre <paulox@paulox.net>

    * version.txt: Updated to 2.1

2007-02-14 Paolo Melchiorre <paulox@paulox.net>

    * version.txt: Updated to 1.0
    * *: Packetized for public release

2006-11-29 Paolo Melchiorre <paulox@paulox.net>

    * tango2plone.sh: Updated tango2plone.sh
    * tango_plone_images: Added new directory
    * version.txt: Updated to 1.0 RC1

2006-11-26 Paolo Melchiorre <paulox@paulox.net>

    * version.txt: Updated to 1.0 beta 1
    * *: Initial version


tango2plone.sh
#!/bin/bash
#plone_images
cp src/gif/16x16/text-html.gif skins/tango_plone_images/document_icon.gif
cp src/gif/16x16/folder.gif skins/tango_plone_images/folder_icon.gif
cp src/gif/32x32/folder.gif skins/tango_plone_images/folder.gif
cp src/gif/16x16/emblem-important.gif skins/tango_plone_images/info_icon.gif
cp src/gif/16x16/next.gif skins/tango_plone_images/linkTransparent.gif
cp src/gif/16x16/mail-forward.gif skins/tango_plone_images/mail_icon.gif
cp src/gif/16x16/document-print.gif skins/tango_plone_images/print_icon.gif
cp src/gif/16x16/system-search.gif skins/tango_plone_images/search_icon.gif
cp src/gif/16x16/folder-saved-search.gif skins/tango_plone_images/topic_icon.gif
cp src/gif/16x16/page-livemarks.gif skins/tango_plone_images/rss.gif
cp src/gif/16x16/x-office-calendar.gif skins/tango_plone_images/event_icon.gif
cp src/gif/16x16/x-office-calendar.gif skins/tango_plone_images/popup_calendar.gif
cp src/gif/16x16/internet-news-reader.gif skins/tango_plone_images/newsitem_icon.gif
cp src/gif/16x16/user.gif skins/tango_plone_images/user.gif
cp src/gif/16x16/group.gif skins/tango_plone_images/group.gif
cp src/gif/16x16/internet-group-chat.gif skins/tango_plone_images/discussionitem_icon.gif
cp src/gif/16x16/view-fullscreen.gif skins/tango_plone_images/fullscreenexpand_icon.gif
cp src/gif/16x16/view-fullscreen-inverted.gif skins/tango_plone_images/fullscreencollapse_icon.gif
cp src/gif/16x16/internet-web-browser.gif skins/tango_plone_images/link_icon.gif
cp src/gif/16x16/document-save.gif skins/tango_plone_images/download_icon.gif
cp src/gif/16x16/edit-copy.gif skins/tango_plone_images/copy_icon.gif
cp src/gif/16x16/edit-cut.gif skins/tango_plone_images/cut_icon.gif
cp src/gif/16x16/edit-delete.gif skins/tango_plone_images/delete_icon.gif
cp src/gif/16x16/edit-paste.gif skins/tango_plone_images/paste_icon.gif
cp src/gif/16x16/preferences-desktop-theme.gif skins/tango_plone_images/skins_icon.gif
cp src/gif/16x16/preferences-system-windows.gif skins/tango_plone_images/navigation_icon.gif
cp src/gif/16x16/text-x-generic-template.gif skins/tango_plone_images/file_icon.gif
cp src/gif/16x16/emblem-favorite.gif skins/tango_plone_images/favorite_icon.gif
cp src/gif/16x16/emblem-favorite.gif skins/tango_plone_images/rtl-favorite_icon.gif
cp src/gif/16x16/image-x-generic.gif skins/tango_plone_images/image_icon.gif
cp src/gif/16x16/preferences-system-windows.gif skins/tango_plone_images/sitemap_icon.gif
cp src/gif/16x16/draw-freehand.gif skins/tango_plone_images/edit.gif
cp src/gif/16x16/draw-freehand.gif skins/tango_plone_images/extedit_icon.gif
cp src/gif/16x16/draw-freehand.gif skins/tango_plone_images/pencil_icon.gif
cp src/gif/extra/defaultUser.gif skins/tango_plone_images/defaultUser.gif

Extensions/Install.py
# -*- coding: utf-8 -*-
#
# File: Install.py
#
# Plango - Copyright (c) 2006-2007 Paolo Melchiorre <paulox@paulox.net>
#
# GNU General Public License (GPL)
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
# 02110-1301, USA.
#

__author__ = """Paolo Melchiorre <paulox@paulox.net>"""
__docformat__ = 'plaintext'


from StringIO import StringIO

from Products.Archetypes.Extensions.utils import install_subskin

from Products.Plango.config import PROJECTNAME
from Products.Plango.config import GLOBALS

def install(self, reinstall=False):
    """ External Method to install Plango """
    out = StringIO()
    print >> out, "Installation log of %s:" % PROJECTNAME
    install_subskin(self, out, GLOBALS)
    return out.getvalue()


def uninstall(self, reinstall=False):
    out = StringIO()
    print >> out, "Uninstallation log of %s:" % PROJECTNAME
    return out.getvalue()


__init__.py
# -*- coding: utf-8 -*-
#
# File: __init__.py
#
# Plango - Copyright (c) 2006-2007 Paolo Melchiorre <paulox@paulox.net>
#
# GNU General Public License (GPL)
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
# 02110-1301, USA.
#

__author__ = """Paolo Melchiorre <paulox@paulox.net>"""
__docformat__ = 'plaintext'


from Products.CMFCore import DirectoryView
GLOBALS = globals()

# Register our skins directory - this makes it available via portal_skins.
DirectoryView.registerDirectory('skins', GLOBALS)
